app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt

```diff
@@ -156,32 +203,94 @@ object MessageController {
         }
     }
 
-    private fun deleteMessages(uids: List<String>, realm: MutableRealm) {
+    fun MutableRealm.createMultiMessagesThreads(messages: List<Message>) {
+        val allThreads = ThreadController.getThreads(realm = this).toMutableList()
+        val threadsToUpsert = mutableListOf<Thread>()
+
+        messages.forEach { message ->
+
+            val messageIds = getMessageIds(message)
+            message.messageIds = messageIds.toRealmSet()
+
+            val thread = allThreads.find { it.messagesIds.intersect(messageIds).isNotEmpty() }
```

Comment:
Couldn't we optimize this `allThreads.find`?

Reply:
Possibly. How?
@LunarX seemed to have some ideas about that.